bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/events/ChannelDescriptionChangedEvent.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.thing.events;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.events.AbstractEvent;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.types.CommandDescription;
+import org.openhab.core.types.StateDescription;
+
+/**
+ * {@link ChannelDescriptionChangedEvent}s will be delivered through the openHAB event bus if the
+ * {@link CommandDescription} or {@link StateDescription} of a channel has changed. Instances must be created with the
+ * {@link ThingEventFactory}.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class ChannelDescriptionChangedEvent extends AbstractEvent {
+
+    public enum CommonChannelDescriptionField {
+        ALL,
```

Comment:
Is ALL necessary (not used I believe) ?

Reply:
It will be needed if a binding publishes the whole `StateDEscription` (e.g. https://github.com/openhab/openhab-addons/pull/10718). Note: I have to change PRs in openHAB add-ons according the new API.